qcodes/instrument_drivers/ZI/ZIUHFLI.py

```diff
@@ -710,6 +710,17 @@ def __init__(self, name: str, device_ID: str, **kwargs) -> None:
                                                 oscs-1, 1, 'freq'),
                                vals=vals.Numbers(0, 600e6))
 
+            self.add_parameter('demod{}_oscillator'.format(oscs),
+                               label='Selected oscillator {}'.format(oscs),
+                               docstring="Connects the demodulator with the "
+                                         "supplied oscillator.",
+                               get_cmd=partial(self._getter, 'demods',
+                                               oscs-1, 0, 'oscselet'),
+                               set_cmd=partial(self._setter, 'demods',
+                                               oscs-1, 0, 'oscselect'),
+                               val_mapping={i+1: i for i in range(8)},
```

Comment:
Should this range also be `number_of_oscillators` ?

Reply:
Yes, it should.

Reply:
same for vals, it's fixed now